### ISSUE TITLE
Difference between Edward and Parmesan

### ISSUE THREAD

**Issue by windweller** [blei-lab/edward#393]
I'm quite new to variational-inference based probabilistic programming (but I have used Church and WebPPL, so I do know a bit about PPL). 

I'm trying to replicate a paper on Stochastic RNN like in this repo: https://github.com/marcofraccaro/srnn

It occurs to me that they built entirely on Theano, Lasagne, and Parmesan (https://github.com/casperkaae/parmesan). 

I want to rewrite it in Tensorflow, and thus able to find Edward. Is there a big difference between Edward and Parmesan (functionality/usage-wise)?

(and off the topic, score function estimator is used to allow backpropagation through a stochastic node, what's the difference between using such approach and using variational inference?)

**Comment by dustinvtran** [comment]
the interface and design of parmesan is very different. like lasagne and keras, it works from a functional perspective. i think this makes sense for neural nets and auto-encoders, although it can be difficult to shoehorn other probability models and non-variational methods in this perspective. 

usage-wise, the variational inference methods for KL minimization should produce the same results.

re:score function estimator. i like to think of variational inference broadly as casting bayesian inference as an optimization problem. given this casting, there are multiple ways to take gradients to actually perform the optimization. for example, the score function estimator is one method to estimate the gradient.

**Comment by windweller** [comment]
@dustinvtran Thank you Dustin! I'm browsing the tutorials but I don't see any example of adding Edward to a neural network like what Parmesan could...is it possible to use Edward to estimate gradient and backprop into a neural network model?

**Comment by dustinvtran** [comment]
you can compare and contrast these two. both implement a variational auto-encoder for binarized MNIST.

+ edward | https://github.com/blei-lab/edward/blob/master/examples/vae.py
+ parmesan | https://github.com/casperkaae/parmesan/blob/master/examples/vae_vanilla.py

**Comment by windweller** [comment]
Thanks! :)
